yellowbox/extras/rabbit_mq.py

```diff
@@ -56,3 +60,44 @@ def enable_management(self):
             raise RuntimeError("Container must have the management port exposed.")
 
         self.container.exec_run("rabbitmq-plugins enable rabbitmq_management")
+
+    @contextmanager
+    def clean_slate(self, force_queue_deletion=False):
+        """
+        Notes:
+            This feature is experimental.
+        """
+        try:
+            management_url = self.management_url()
+        except RuntimeError as e:
+            raise RuntimeError('management must be enabled for the service for clean_slate') from e
+
+        queues_url = management_url + 'api/queues'
+        response = requests.get(queues_url, auth=(self.user, self.password))
```

Comment:
def _get_queues(self)?

Reply:
seems like a lot of overhead for something that will only be called once

Reply:
🤷 might be useful later on. The only thing I'm not sure about is again, whether we should have utility functions on the service or not. Still debating that one.

Reply:
it's experimantal

Reply:
Not talking about "clean slate". Talking about generally adding application functions to the Service object.